etcdutl/snapshot/v3_snapshot.go

```diff
@@ -138,31 +139,33 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
 		if v != nil {
 			ds.Version = v.String()
 		}
+		var kv mvccpb.KeyValue
 		c := tx.Cursor()
 		for next, _ := c.First(); next != nil; next, _ = c.Next() {
 			b := tx.Bucket(next)
-			if b == nil {
-				return fmt.Errorf("cannot get hash of bucket %s", string(next))
-			}
-			if _, err = h.Write(next); err != nil {
-				return fmt.Errorf("cannot write bucket %s : %v", string(next), err)
-			}
+			_, _ = h.Write(next)
+
 			iskeyb := (string(next) == "key")
 			if err = b.ForEach(func(k, v []byte) error {
-				if _, herr := h.Write(k); herr != nil {
-					return fmt.Errorf("cannot write to bucket %s", herr.Error())
-				}
-				if _, herr := h.Write(v); herr != nil {
-					return fmt.Errorf("cannot write to bucket %s", herr.Error())
-				}
+				_, _ = h.Write(k)
+				_, _ = h.Write(v)
 				if iskeyb {
-					rev := mvcc.BytesToRev(k)
+					var rev mvcc.BucketKey
+					rev, err = mvcc.BytesToBucketKey(k)
```

Comment:
Please use `mvcc.BytesToRev` instead of `mvcc.BytesToBucketKey`.  Note using `mvcc.BucketKey` only makes sense when you need to know the the revision is persisted on disk.

FYI. BucketKey is only used when removing key. 

Actually I tend to remove `BucketKey` completely to simplify the implementation. But it's a separate topic.